File: src/test/kotlin/me/pcasaes/hexoids/core/domain/model/EntityIdTest.java
package me.pcasaes.hexoids.core.domain.model;

import org.junit.jupiter.api.Test;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;

class EntityIdTest {

    @Test
    void testConversion() {
        var uuid = UUID.randomUUID();

        var entityId = EntityId.of(uuid);

        var guid = entityId.getGuid();

        var entityId2 = EntityId.of(guid);

        var uuid2 = entityId2.getId();

        assertEquals(uuid, uuid2);
    }
}

File: src/test/kotlin/me/pcasaes/hexoids/core/domain/model/EntityIdTest.kt
package me.pcasaes.hexoids.core.domain.model

import me.pcasaes.hexoids.core.domain.model.EntityId.Companion.of
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.UUID

class EntityIdTest {
    @Test
    fun testConversion() {
        val uuid = UUID.randomUUID()

        val entityId = of(uuid)

        val guid = entityId.getGuid()

        val entityId2 = of(guid)

        val uuid2 = entityId2.getId()

        Assertions.assertEquals(uuid, uuid2)
    }
}